### ISSUE TITLE
i am getting this error even when im running it on google cloud compute virtual machine engine. does this impact the performance and weights assigned in my model as the model still runs but gives extemely poor results . Also , in order to address class imbalance in my keras ANN model in introduced the class_weights option is there any other way to handle that situation ??

### ISSUE THREAD

**Issue by NTomarRajput** [rstudio/keras#854]
i am getting this error even when im running it on google cloud compute virtual machine engine. does this impact the performance and weights assigned in my model as the model still runs but gives extemely poor results . Also , in order to address class imbalance in my keras ANN model in introduced the class_weights option is there any other way to handle that situation ??

_Originally posted by @NTomarRajput in https://github.com/rstudio/keras/issues/444#issuecomment-516662003_

**Comment by dfalbel** [comment]
Could you please copy and paste the error message you are seeing and the code you are running? Without any specific information I can't say what it can affect.

**Comment by NTomarRajput** [comment]
library(tensorflow)
library(caTools)
final<-read.csv('FinaldatamatrixFnM.csv')
final <-final[,-1]

set.seed(2019)
ind<-sample(1:3,nrow(final), replace = T, prob= c(0.7,0.15,0.15))
training <-final[ind==1,c(2:2784)]
trainingtarget<-final[ind==1,1]
test<-final[ind==2,c(2:2784)]
testtarget<-final[ind==2,1]
validation <-final[ind==3,c(2:2784)]
validation.target<-final[ind==3,1]

traininglabels=to_categorical(trainingtarget)
testlabels=to_categorical(testtarget)
validationlabels=to_categorical(validation.target)

set.seed(2019)
model_keras <- keras_model_sequential()

model_keras %>% 
  layer_dense (units              = 62, #=> Num Of Nodes
               activation         = "relu",    
               input_shape        = 2783) %>% 
  layer_dropout (rate = 0.3) %>%  #=> Dropout Below 10%: Prevent overfittin
  # (3) Output Layer-----------------------------------------------------

  layer_dense (units              = 35, #=> Num Of Nodes
               activation         = "relu") %>% 
  layer_dropout (rate = 0.2) %>%  
  layer_dense (units              = 16, #=> Num Of Nodes
               activation         = "relu") %>% 
  layer_dropout (rate = 0.1) %>%  #=> Dropout Below 10%: Prevent over
  layer_dense (units              = 16, #=> Num Of Nodes
               activation         = "relu")%>% 
layer_dropout (rate = 0.01) %>%
layer_dense (units              = 2, #=> Binary/Multi?=>That Number
             
             activation         = "softmax") %>% #=> Common for Binary
  # (4) Compile Model-----------------------------------------------------
compile (optimizer = 'rmsprop', #=> Most Popular for Optimization Algo.
         loss      = 'binary_crossentropy', #=> Binary Classification
         metrics   = c('accuracy') ) #=> Train/Test Evaluation

# Check

model_keras

history <- fit (
  object           = model_keras,             # => Our Model
  x                = as.matrix(training), #=> Matrix
  y                = traininglabels,             #=> Numeric Vector 
  batch_size       = 50,     #=> #OfSamples/gradient update in each epoch
  epochs           = 20,     #=> Control Training cycles
  validation_split = 0.30,
class_weight = list("0"=1,"1"=0.22) #61407(observation in class '1')  282213(observation in class'0')
                                      # therefore 61407/282213= 0.217591=~0.22
) 

model_keras %>% evaluate(as.matrix(validation), validationlabels)
pred<-model_keras %>% predict_classes(as.matrix(validation))
pred1<-model_keras %>% predict_classes(as.matrix(test))
table(Predicted=pred, Actual=validation.target)
table(Predicted=pred1, Actual=testtarget)

prob<-model_keras%>%predict_proba(as.matrix(validation))
prob2<-model_keras%>%predict_proba(as.matrix(test))
valresult<-cbind.data.frame(pred,prob,validation.target)
testresult2<-cbind.data.frame(pred1,prob2,testtarget)

write.csv(valresult,'krvalfmresult.csv')
write.csv(testresult2,'krtestfmresultest.csv')

**Comment by dfalbel** [comment]
Can you please post the error message you are seeing?
